.github/workflows/ci.yml

```diff
@@ -148,3 +150,32 @@ jobs:
           disable_safe_directory: true
           files: lcov.info
           token: ${{ secrets.CODECOV_TOKEN }}
+
+  release:
+    if: startsWith(github.ref, 'refs/tags/')
+    name: Create new release
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            flags: --target x86_64-unknown-linux-gnu
+            asset_name: "uhyve-${{ github.ref_name }}-linux-amd64"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+      - name: Build
+        run: cargo build --release --locked ${{ matrix.flags }}
+      - name: Upload binaries to new release
+        id: release
+        # v2.9.0, pinned to the corresponding commit hash for hardening reasons
+        uses: svenstaro/upload-release-action@04733e069f2d7f7f0b4aebc4fbdbce8613b03ccd
```

Comment:
I'd prefer to use GitHub CLI instead, which is installed by default on GitHub-hosted runners and the preferred way to interact with GitHub in GitHub actions.

Reply:
I did look at your example and consider that, can't exactly remember why I did not opt for using the GitHub CLI instead. I'll take a look at the commit log and some notes and get back to this.